front/pages/api/v1/w/[wId]/data_sources/[dsId]/documents/[documentId]/index.ts

```diff
@@ -212,18 +212,23 @@ async function handler(
   >,
   auth: Authenticator
 ): Promise<void> {
-  const { name } = req.query;
-  if (typeof name !== "string") {
+  const { dsId } = req.query;
+  if (typeof dsId !== "string") {
     return apiError(req, res, {
       status_code: 404,
       api_error: {
-        type: "data_source_not_found",
-        message: "The data source you requested was not found.",
+        type: "invalid_request_error",
+        message: "Invalid path parameters.",
       },
     });
   }
 
-  const dataSource = await getDataSource(auth, name);
+  const dataSource = await DataSourceResource.fetchByNameOrId(
+    auth,
+    dsId,
+    // TODO(DATASOURCE_SID): Clean-up
```

Comment:
nit: what cleanup would be needed here? change the call to "fetchById" when we remove "name"? IMO we can remove the com (will be done auto by IDE renaming)